components/style/values/specified/svg.rs

```diff
@@ -260,3 +261,18 @@ impl ToCss for SVGPaintOrder {
         Ok(())
     }
 }
+
+/// Specified MozContextProperties value.
+/// Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties)
+pub type MozContextProperties = CustomIdent;
+
+impl Parse for MozContextProperties {
```

Comment:
This is implementing `parse` for `CustomIdent`, which is intentionally not implemented.

So we should make `MozContextProperties` a different type, like:

```rust
pub struct MozContextProperties(pub CustomIdent);
```

And implement `parse` on that.

Reply:
(In particular, not all `CustomIdent`s should be parsed like `-moz-context-properties`.

Reply:
I'm getting a lot of errors when trying to make `MozContextProperties` a different type. I suppose it could be related to the `match_ignore_ascii_case!()` macro in the `from_ident()` function.

`./mach build-geckolib`:
```Rust
error[E0308]: mismatched types
     --> /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko/atom_macro.rs:18554:60
      |
18554 |   {{ #[allow(unsafe_code)] #[allow(unused_unsafe)]unsafe { $crate::string_cache::atom_macro::atom_from_static($crate::string_cache::atom_macro::nsGkAtoms_fill as *mut _) } }};
      |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `values::CustomIdent`, found struct `gecko_string_cache::Atom`
      | 
     ::: /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko_properties.rs
      |
20499 |             if servo.0 == atom!("fill") {
      |                           ------------- in this macro invocation
      |
      = note: expected type `values::CustomIdent`
                 found type `gecko_string_cache::Atom`

error[E0308]: mismatched types
     --> /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko/atom_macro.rs:18840:60
      |
18840 |   {{ #[allow(unsafe_code)] #[allow(unused_unsafe)]unsafe { $crate::string_cache::atom_macro::atom_from_static($crate::string_cache::atom_macro::nsGkAtoms_stroke as *mut _) } }};
      |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `values::CustomIdent`, found struct `gecko_string_cache::Atom`
      | 
     ::: /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko_properties.rs
      |
20501 |             } else if servo.0 == atom!("stroke") {
      |                                  --------------- in this macro invocation
      |
      = note: expected type `values::CustomIdent`
                 found type `gecko_string_cache::Atom`

error[E0308]: mismatched types
     --> /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko/atom_macro.rs:18556:60
      |
18556 |   {{ #[allow(unsafe_code)] #[allow(unused_unsafe)]unsafe { $crate::string_cache::atom_macro::atom_from_static($crate::string_cache::atom_macro::nsGkAtoms_fill_opacity as *mut _) } }};
      |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected str
      | 
     ::: /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko_properties.rs
      |
20503 |             } else if servo.0 == atom!("fill-opacity") {
      |                                  --------------------- in this macro invocation
      |
      = note: expected type `values::CustomIdent`
                 found type `gecko_string_cache::Atom`

error[E0308]: mismatched types
     --> /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko/atom_macro.rs:18852:60
      |
18852 |   {{ #[allow(unsafe_code)] #[allow(unused_unsafe)]unsafe { $crate::string_cache::atom_macro::atom_from_static($crate::string_cache::atom_macro::nsGkAtoms_stroke_opacity as *mut _) } }};
      |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected s
      | 
     ::: /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko_properties.rs
      |
20505 |             } else if servo.0 == atom!("stroke-opacity") {
      |                                  ----------------------- in this macro invocation
      |
      = note: expected type `values::CustomIdent`
                 found type `gecko_string_cache::Atom`

error[E0599]: no method named `into_addrefed` found for type `values::CustomIdent` in the current scope
     --> /home/user/dev/rust/servo/target/geckolib/debug/build/style-b27e42e5c2727575/out/gecko_properties.rs:20508:37
      |
20508 |             gecko.mRawPtr = servo.0.into_addrefed();
      |                                     ^^^^^^^^^^^^^

error: aborting due to 5 previous errors

error: Could not compile `style`.
```

Reply:
That's because those `servo.0` now are a `CustomIdent` instead of an `Atom`. You should be able to use `(servo.0).0` or (I think, if I recall correctly), `servo.0.as_atom()`.

Reply:
`servo.0.as_atom()` isn't implemented, but `(servo.0).0` worked. Please, have a look at the current changes. If they're okay, I'll `squash` and `push` the changes.